src/java/org/apache/cassandra/service/DataResolver.java

```diff
@@ -56,10 +58,16 @@ public class DataResolver extends ResponseResolver
     private final boolean enforceStrictLiveness;
 
     DataResolver(Keyspace keyspace, ReadCommand command, ConsistencyLevel consistency, int maxResponseCount, long queryStartNanoTime)
+    {
+        this(keyspace, command, consistency, maxResponseCount, queryStartNanoTime, command.metadata().enforceStrictLiveness());
+    }
+
+    @VisibleForTesting
+    DataResolver(Keyspace keyspace, ReadCommand command, ConsistencyLevel consistency, int maxResponseCount, long queryStartNanoTime, boolean enforceStrictLiveness)
```

Comment:
ReadCommand is quite hard to mock / initialize in test, it just needs too much stuff. When command is null, for the sake of the test to have something there for `DataResolver` constructor, it would fail on `command.metadata().enforceStrictLiveness`. So I did one more constructor which might be called from the test where we set it without going through (null) `command`.